File: Day-03/solution.js
import readline from 'readline';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Enter a string: ', (inputStr) => {
    if (inputStr.trim().length === 0) {
        console.log("Empty string!");
        rl.close();
        return;
    }

    const vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];

    const vowelCount = inputStr.split('').reduce(
        (count, char) => {
            return count + (vowels.includes(char) ? 1 : 0);
        },
        0
    );

    console.log(`Number of vowels: ${vowelCount}`);

    rl.close();
});

File: Day-03/solution.py
from functools import reduce

input_str = input('Enter a string: ')

if not input_str.strip():
    print("Empty string!")
    exit()

vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']

vowel_count = reduce(lambda acc, char: acc +
                    (1 if char in vowels else 0), list(input_str), 0)

print(f"Number of vowels: {vowel_count}")
